app/models/user.rb

```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  validates :name, :email, :password, :phone, :user_name, :role, presence: true
+  validates :email, format: {with: URI::MailTo::EMAIL_REGEXP}
+  validates :email, :user_name, uniqueness: {message: 'has already been taken'}
+  enum role: [:user, :admin]
```

Comment:
this is fine. Normally we follow the below format for defining enum fields  `enum role:  { user: 0, admin: 1 }`.
Thank you.

Reply:
Thanks for mentioning apu. changed it as mentioned.